### ISSUE TITLE
Request for more SFT training details for reproducibility

### ISSUE THREAD

**Issue by jack0rich** [arvillion/ActiveVLN#28]
Hi, thank you for releasing the ActiveVLN codebase and checkpoints.

I am currently trying to reproduce the SFT stage before RL training, but the reproduced SFT performance is much worse than expected. I noticed that the README provides only a brief “For SFT Reference” section, but does not include the SFT codebase or detailed training settings. Since the RL stage is built on top of the SFT checkpoint, I believe these details are very important for reproducibility.

Could you please provide more details about the SFT setup, if possible? In particular, I would really appreciate clarification on the following points:

1. Dataset construction
- How exactly was the SFT training data constructed from R2R / R2R-EnvDrop / RxR?
- What is the exact conversation or sample format used for training?
- How are multi-turn trajectories organized?
- For RxR, how was the dataset converted into the R2R action space in practice?

2. SFT framework and training recipe
- Which SFT framework did you use (e.g., transformers, LLaMA-Factory, or another internal framework)?
- Did you use full-parameter finetuning or LoRA?
- What were the key hyperparameters (batch size, learning rate, number of epochs/steps, max sequence length, image resolution, etc.)?

3. LoRA configuration
- If LoRA was used, what were the LoRA settings (rank, alpha, dropout, target modules, etc.)?

4. Loss design
- Did you apply any special loss weighting during SFT?
- For example, were special tokens or action tokens (such as navigation actions / stop token / end-of-turn tokens) assigned larger weights?

5. Prompt / template details
- Could you share the exact system prompt or training template used in SFT?
- Are there any special formatting details that are critical for reproducing the reported SFT checkpoint quality?

At the moment, my reproduced SFT model performs significantly worse, so I suspect there are some important implementation details missing from the current release. Any additional guidance, config files, or even partial training scripts would be extremely helpful.

Thanks again for your work and for making the project public.

**Comment by waynechu1021** [comment]
> Hi, thank you for releasing the ActiveVLN codebase and checkpoints.
> 
> I am currently trying to reproduce the SFT stage before RL training, but the reproduced SFT performance is much worse than expected. I noticed that the README provides only a brief “For SFT Reference” section, but does not include the SFT codebase or detailed training settings. Since the RL stage is built on top of the SFT checkpoint, I believe these details are very important for reproducibility.
> 
> Could you please provide more details about the SFT setup, if possible? In particular, I would really appreciate clarification on the following points:
> 
> 1. Dataset construction
> 
> * How exactly was the SFT training data constructed from R2R / R2R-EnvDrop / RxR?
> * What is the exact conversation or sample format used for training?
> * How are multi-turn trajectories organized?
> * For RxR, how was the dataset converted into the R2R action space in practice?
> 
> 2. SFT framework and training recipe
> 
> * Which SFT framework did you use (e.g., transformers, LLaMA-Factory, or another internal framework)?
> * Did you use full-parameter finetuning or LoRA?
> * What were the key hyperparameters (batch size, learning rate, number of epochs/steps, max sequence length, image resolution, etc.)?
> 
> 3. LoRA configuration
> 
> * If LoRA was used, what were the LoRA settings (rank, alpha, dropout, target modules, etc.)?
> 
> 4. Loss design
> 
> * Did you apply any special loss weighting during SFT?
> * For example, were special tokens or action tokens (such as navigation actions / stop token / end-of-turn tokens) assigned larger weights?
> 
> 5. Prompt / template details
> 
> * Could you share the exact system prompt or training template used in SFT?
> * Are there any special formatting details that are critical for reproducing the reported SFT checkpoint quality?
> 
> At the moment, my reproduced SFT model performs significantly worse, so I suspect there are some important implementation details missing from the current release. Any additional guidance, config files, or even partial training scripts would be extremely helpful.
> 
> Thanks again for your work and for making the project public.

Hello, thank you for your interest in ActiveVLN. I am the collaborator of ActiveVLN, primarily responsible for the SFT training. Below, I will answer your questions.

1. Regarding dataset construction:

+ For the action label structure, you can refer to the [NaVIDA](https://github.com/waynechu1021/NAVIDA/tree/main/src/data) and modify it for multi-turn dialogue.

+ For the RxR dataset, we used a version processed by StreamVLN, which is already aligned with the R2R action space: https://huggingface.co/datasets/cywan/StreamVLN-Trajectory-Data/tree/c91703117f39f1566497b003322ac0399f916554/RxR


2. Regarding SFT training:

+ For the training framework, we use the transformers library. You can also use LLaMA-Factory or other frameworks; it's all the same.

+ We did not use LoRA. We froze the vision encoder while keeping the projector and LLM backbone involved in training.

+ Hyperparameters depend on your specific training device. For us, we trained for 1 epoch using a learning rate of 2e-5, with the image resolution set to 
```
processor.image_processor.max_pixels = 76800 
processor.image_processor.min_pixels = 1024
```

3. As mentioned above, we did not use LoRA.

4. We did not have a special loss design; we only calculated the next token prediction loss for the response portion of each epoch.

5. You can refer to the eval section of the code for the Prompt template.

**Comment by jack0rich** [comment]
Thank you very much for your detailed reply. I really appreciate you taking the time to clarify these points.Your explanation is very helpful, I will carefully follow these settings.
